src/common/views.py

```diff
@@ -230,8 +252,36 @@ def show_model_form(self, request, id: int = None, edit: str = "", field: str =
             else self.table_view.model_relation_fields[relation_field_name][-2]
         )
 
+        # add and remove urls for formset
+        if len(self.formset_options) > 0:
+            add_formset_url = (
+                reverse(
+                    f"{self.model_name.lower()}_formset",
+                    args=[formset_count + 1],
+                )
+                if self.cud_actions_rule
+                else "#"
+            )
+
+            remove_formset_url = (
+                reverse(
+                    f"{self.model_name.lower()}_formset",
+                    args=[formset_count - 1],
+                )
+                if self.cud_actions_rule
+                else "#"
+            )
+
         context = {
             "form": form,
+            "formset_title": self.formset_options[2] if self.formset_options else None,
+            "formset_form": self.model_formset() if self.model_formset else None,
+            "add_formset_url": add_formset_url
+            if "add_formset_url" in locals()
+            else "#",
```

Comment:
Are we assigning "#" to URLs? We might need a different approach to non-existing URLs.
If we use this approach now, this might be an issue we need to fix in the future, so we might as well address it now.

One idea is to handle this in the frontend - maybe we don't use this context data in the template when it doesn't exist?